src/systems/media-frames.js

```diff
@@ -0,0 +1,219 @@
+      // TODO would be nice to fix the timing so we don't need to check this
+      if (!this.physicsSystem.bodyInitialized(bodyUUID)) continue;
+
+      const objectToCapture = getCapturableObjectCollidingWithBody(this.physicsSystem, zone.data.mediaType, bodyUUID);
+
+      const previewMesh = zone.el.getObject3D("preview");
+      const guideMesh = zone.el.getObject3D("guide");
+
+      guideMesh.visible = this.interactionSystem.isHoldingAnything(isCapturableByType[zone.data.mediaType]);
+
+      if (zone.state === TARGET_ZONE_STATE.EMPTY) {
+        // TODO not sure if we want to use tags for this
+        if (objectToCapture && !isTagged(objectToCapture.el, "isHoveringMediaTargetZone")) {
+          setTag(objectToCapture.el, "isHoveringMediaTargetZone");
+
+          zone.state = TARGET_ZONE_STATE.HOVERING;
+          zone.target = objectToCapture;
+
+          const imageObject3D = objectToCapture.el.getObject3D("mesh");
+
+          // TODO this feels a bit hacky and error prone, also needs support for previewing 3D objects
+          if (isCapturableByType[MediaType.ALL_2D]) {
+            if (imageObject3D.material) {
+              previewMesh.material.map = imageObject3D.material.map;
+            }
+            previewMesh.material.needsUpdate = true;
+            previewMesh.scale.copy(imageObject3D.scale);
+            previewMesh.scale.multiplyScalar(scaleForAspectFit(zone.data.bounds, imageObject3D.scale));
+            // Preview mesh UVs are set to accomidate textureLoader default, but video textures don't match this
+            previewMesh.scale.y *= TEXTURES_FLIP_Y !== previewMesh.material.map.flipY ? -1 : 1;
+            previewMesh.matrixNeedsUpdate = true;
+            previewMesh.visible = true;
+          }
+
+          guideMesh.material.uniforms.color.value.set(0x6fc0fd);
+        }
+      } else {
+        // TODO currently this works because the object will continue to collide, we may not want this
+        if (!objectToCapture) {
+          // TODO these cmp.target.el are to handle object being destroyed while captured, but they feel a bit ugly
+          zone.target.el && setTag(zone.target.el, "isHoveringMediaTargetZone", false);
+
+          if (zone.state === TARGET_ZONE_STATE.FULL) {
+            zone.target.scale.copy(zone.originalTargetScale);
+            zone.target.matrixNeedsUpdate = true;
+            zone.target.el && zone.target.el.components["floaty-object"].setLocked(false);
+          }
+
+          zone.state = TARGET_ZONE_STATE.EMPTY;
+          zone.target = null;
+
+          previewMesh.material.map = null;
+          previewMesh.material.needsUpdate = true;
+          previewMesh.visible = false;
+
+          guideMesh.material.uniforms.color.value.set(0x2f80ed);
+        }
+
+        if (zone.state === TARGET_ZONE_STATE.FULL && zone.data.dynamic) {
+          zone.target.position.copy(zone.el.object3D.position);
+          zone.target.rotation.copy(zone.el.object3D.rotation);
```

Comment:
I expect zone targets to be at root level of the scene graph, but zones may be deeper in the hierarchy (e.g. when added in blender). Might want `worldPosition/Rotation`.